### PR TITLE
feat: collecting messaging.* span attributes in botocore instrumentation

### PR DESCRIPTION
**Pull request by saartochner**
# Description

Support messaging system in the sqs extension. 
Following spec: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [V] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [V] Testing using moto to check all the three API cases: SendMessage, SendMessageBatch, ReceiveMessage

# Does This PR Require a Core Repo Change?

- [V] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [V] Followed the style guidelines of this project
- [V] Changelogs have been updated
- [V] Unit tests have been added
- [V] Documentation has been updated

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/extensions/sqs.py [hunk 1]
```diff
@@ -24,3 +29,27 @@ def extract_attributes(self, attributes: _AttributeMapT):
         if queue_url:
             # TODO: update when semantic conventions exist
             attributes["aws.queue_url"] = queue_url
+            attributes[SpanAttributes.MESSAGING_SYSTEM] = "aws.sqs"
+            attributes[SpanAttributes.MESSAGING_URL] = queue_url
+            attributes[SpanAttributes.MESSAGING_DESTINATION] = queue_url.split(
+                "/"
+            )[-1]
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/extensions/sqs.py:36]: wrap this in try-except. The instrumentation should neve throw any exceptions.

File: instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/extensions/sqs.py [hunk 2]
```diff
@@ -24,3 +29,27 @@ def extract_attributes(self, attributes: _AttributeMapT):
...
+                )
+            elif operation == "SendMessageBatch" and result.get("Successful"):
+                span.set_attribute(
+                    SpanAttributes.MESSAGING_MESSAGE_ID,
+                    result["Successful"][0]["MessageId"],
+                )
+            elif operation == "ReceiveMessage" and result.get("Messages"):
+                span.set_attribute(
+                    SpanAttributes.MESSAGING_MESSAGE_ID,
+                    result["Messages"][0]["MessageId"],
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/extensions/sqs.py:54]: Same as above, wrap in try-except block to ignore the potential `KeyError` or `IndexError`

**Comment by srikanthccv** [comment]
Also sign the CLA

File: CHANGELOG.md [hunk 3]
```diff
@@ -20,6 +20,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `opentelemetry-instrumentation-django` Fixed bug where auto-instrumentation fails when django is installed and settings are not configured.
   ([#1369](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1369))
 - `opentelemetry-instrumentation-system-metrics` add supports to collect system thread count. ([#1339](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1339))
+- `opentelemetry-instrumentation-botocore` add support for `messaging.*` in the sqs extension.
+  ([#1350](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1350))
```
Comment (srikanthccv) [inline CHANGELOG.md:24]: Please move this to Unreleased section

File: instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/extensions/sqs.py [hunk 4]
```diff
@@ -11,11 +11,19 @@
...
     _AttributeMapT,
     _AwsSdkExtension,
+    _BotoResultT,
 )
+from opentelemetry.semconv.trace import SpanAttributes
+from opentelemetry.trace.span import Span
+
+_SUPPORTED_OPERATIONS = ["SendMessage", "SendMessageBatch", "ReceiveMessage"]
+
+logger = logging.getLogger(__name__)
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/extensions/sqs.py:26]:
```suggestion
_logger = logging.getLogger(__name__)
```

File: instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/extensions/sqs.py [hunk 5]
```diff
@@ -24,3 +32,38 @@ def extract_attributes(self, attributes: _AttributeMapT):
...
+                        SpanAttributes.MESSAGING_MESSAGE_ID,
+                        result["Successful"][0]["MessageId"],
+                    )
+                elif operation == "ReceiveMessage" and result.get("Messages"):
+                    span.set_attribute(
+                        SpanAttributes.MESSAGING_MESSAGE_ID,
+                        result["Messages"][0]["MessageId"],
+                    )
+            except (IndexError, KeyError):
+                logger.error("Could not extract the messaging message if")
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/extensions/sqs.py:69]:
```suggestion
                logger.error("Could not extract the messaging message ID")
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,6 +6,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
+- `opentelemetry-instrumentation-botocore` add support for `messaging.*` in the sqs extension.
+  ([#1350](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1350))
```
Comment (srikanthccv) [inline CHANGELOG.md:11]:
```suggestion

### Added
- `opentelemetry-instrumentation-botocore` add support for `messaging.*` in the sqs extension.
  ([#1350](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1350))

```

**Review by srikanthccv — APPROVED** [review]
Please sign the CLA

**Comment by saartochner** [comment]
Hi @srikanthccv,
I signed the CLA, can you please approve the workflow?
Thanks!

**Pull request merged**